torch/csrc/distributed/autograd/context/context.h

```diff
@@ -71,6 +71,7 @@ class TORCH_API DistAutogradContext {
   friend class BackwardPassCleanupGuard;
   friend class DistEngine;
   friend class RecvRpcBackward;
+  friend class AccumulateGradPreHook;
```

Comment:
nit: DistAccumulateGradCaptureHook

Reply:
Actually the execution of this hook is not tied to output capture in the local engine. That means a hook can be installed and executed even if an `AccumulateGrad` node doesn't have `captures_` set  in`exec_info_`.

Reply:
@albanD @pritamdamania87 
Should we stop capturing the grads for output nodes in dist_engine? My understanding is capturing the grads is for accumulating them into RPC context in the graph task callback. `graphTask->init_to_execute(dummyRoot, outputEdges);` call does the trick to tell the local engine to capture grads for `outputEdges`.